lib/modifications.rb

```diff
@@ -193,6 +211,138 @@ def self.modify!(results)
+        pulse_ox_clone.component.last.valueQuantity = FHIR::Quantity.new
+        pulse_ox_clone.component.last.valueQuantity.value = 40
+        pulse_ox_clone.component.last.valueQuantity.unit = '%'
+        pulse_ox_clone.component.last.valueQuantity.system = 'http://unitsofmeasure.org'
+        pulse_ox_clone.component.last.valueQuantity.code = '%'
+        # add the Pulse Oximetry as a new Bundle entry
+        selection_device.entry << create_bundle_entry(pulse_ox_clone)
+        # add the Pulse Oximetry into the provenance
+        provenance.resource.target << FHIR::Reference.new
+        provenance.resource.target.last.reference = "urn:uuid:#{pulse_ox_clone.id}"
+        puts "  - Cloned Pulse Oximetry and Added Components: #{selection_pulse_ox.entry.first.resource.id}"
+      end
+
+      # Observation Data Absent Reasons
+      # observation_profiles_valueQuantity_required = [
+      #   'http://hl7.org/fhir/us/core/StructureDefinition/pediatric-bmi-for-age',
+      #   'http://hl7.org/fhir/us/core/StructureDefinition/pediatric-weight-for-height',
+      #   'http://hl7.org/fhir/StructureDefinition/bmi',
+      # ]
+      observation_profiles_valueCodeableConcept_required = [
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-smokingstatus'
+      ]
+      observation_profiles_components_required = [
+        'http://hl7.org/fhir/StructureDefinition/bp'
+      ]
+      observation_profiles = [
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-observation-lab',
+        # 'http://hl7.org/fhir/us/core/StructureDefinition/pediatric-bmi-for-age',
+        # 'http://hl7.org/fhir/us/core/StructureDefinition/pediatric-weight-for-height',
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-smokingstatus',
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-pulse-oximetry',
+        'http://hl7.org/fhir/StructureDefinition/resprate',
+        'http://hl7.org/fhir/StructureDefinition/heartrate',
+        'http://hl7.org/fhir/StructureDefinition/bodytemp',
+        'http://hl7.org/fhir/StructureDefinition/bodyheight',
+        'http://hl7.org/fhir/StructureDefinition/headcircum',
+        'http://hl7.org/fhir/StructureDefinition/bodyweight',
+        # 'http://hl7.org/fhir/StructureDefinition/bmi',
+        'http://hl7.org/fhir/StructureDefinition/bp'
+      ]
+      puts "  - Processing Observation Data Absent Reasons"
+      results.each do |bundle|
+        break if observation_profiles.empty?
+        observation_profiles.delete_if do |profile_url|
+          entry = bundle.entry.find {|e| e.resource.resourceType == 'Observation' && e.resource.meta&.profile&.include?(profile_url) }
+          if entry
+            instance = entry.resource
+            instance.dataAbsentReason = create_codeable_concept('http://terminology.hl7.org/CodeSystem/data-absent-reason', 'unknown', 'Unknown')
+            # if observation_profiles_valueQuantity_required.include?(profile_url)
```

Comment:
Is this `DATAABSENTREASONEXTENSIONGOESHERE` not needed anymore, or an oversight in checking in?

Reply:
It is not needed anymore, but I purposefully left it in here, in case we find out that we need to do this, or something similar. I don't want to have to recreate the code if that comes to pass.